app/src/main/java/kth/jjve/moveable/MainActivity.java

```diff
@@ -427,45 +432,49 @@ public void savingCancelled() {
 
             // rotation from acceleration data
-            roll_x = Math.toDegrees(Math.atan(ay / (Math.sqrt(Math.pow(ax, 2) + Math.pow(ax, 2)))));
-            pitch_y = Math.toDegrees(Math.atan(ax / (Math.sqrt(Math.pow(ay, 2) + Math.pow(ax, 2)))));
-            yaw_z = Math.toDegrees(Math.atan( (Math.sqrt(Math.pow(ax, 2) + Math.pow(ay, 2)) / az )));
+            pitchRollYaw = DataProcess.rotFromAcc(ax, ay, az);
         }
-        //Log.i(LOG_TAG, "Acceleration    x: " + ax + ", y: " + ay + ", z: " + az);
-        //Log.i(LOG_TAG, "roll_x: " + roll_x + ", pitch_y: " + pitch_y + ", yaw_z: " + yaw_z);
+        tempIntRotAcc.setText("Rotation from accelerometer: " + Math.round(pitchRollYaw[0]) + ", " + Math.round(pitchRollYaw[1]) + ", " + Math.round(pitchRollYaw[2]));
+
+        EWMA_filtered_values[0] = DataProcess.EMWA_filter(pitchRollYaw[0], (float) 0.5, EWMA_filtered_values[0]);
+        EWMA_filtered_values[1] = DataProcess.EMWA_filter(pitchRollYaw[1], (float) 0.5, EWMA_filtered_values[1]);
+        EWMA_filtered_values[2] = DataProcess.EMWA_filter(pitchRollYaw[2], (float) 0.5, EWMA_filtered_values[2]);
+        Log.i(LOG_TAG, "EMWA filter: " + EWMA_filtered_values[0] + ", " + EWMA_filtered_values[1] + ", " + EWMA_filtered_values[2]);
 
         if (event.sensor.getType()==Sensor.TYPE_GYROSCOPE){
-            gx=event.values[0];
-            gy=event.values[1];
-            gz=event.values[2];
+            float gx = (float) Math.toDegrees(event.values[0]);
+            float gy= (float) Math.toDegrees(event.values[1]);
+            float gz= (float) Math.toDegrees(event.values[2]);
+
+            dT = (System.currentTimeMillis() - timestamp) / (float) 1000; //units: seconds
+            timestamp = System.currentTimeMillis(); // for storing old value
 
             //rotation from gyroscope
-            rot_x = rotFromGyroscope(gx, rot_x);
-            rot_y = rotFromGyroscope(gx, rot_y);
-            rot_z = rotFromGyroscope(gx, rot_z);
+            rotFromGyro = DataProcess.rotFromGyroscope(gx, gy, gz, rotFromGyro[0], rotFromGyro[1], rotFromGyro[2], dT);
         }
-        //Log.i(LOG_TAG, "Gyroscope    x: " + gx + ", y: " + gy + ", z: " + gz);
-        Log.i(LOG_TAG, "Rotation:      rot_x: " + rot_x + ", rot_y: " + rot_y + ", rot_z: " + rot_z);
+
+        tempIntRotGyro.setText("Rotation from gyroscope: " + Math.round(rotFromGyro[0]) + ", " + Math.round(rotFromGyro[1]) + ", " + Math.round(rotFromGyro[2]));
+
+        //Complimentary filter
+        complimentary_filtered_values[0] = DataProcess.complimentaryFilter(rotFromGyro[0], pitchRollYaw[0], (float) 0.5, complimentary_filtered_values[0], dT);
+        complimentary_filtered_values[1] = DataProcess.complimentaryFilter(rotFromGyro[1], pitchRollYaw[1], (float) 0.5, complimentary_filtered_values[1], dT);
+        complimentary_filtered_values[2] = DataProcess.complimentaryFilter(rotFromGyro[2], pitchRollYaw[2], (float) 0.5, complimentary_filtered_values[2], dT);
```

Comment:
I don't see why you need the lines for this. You could clean it up majorly.

Reply:
Again... temporary values until I validate the angles